packages/gel/src/options.ts

```diff
@@ -427,7 +432,9 @@ export class Options {
       this.config.size === 0 &&
       this.globals.size === 0 &&
       this.moduleAliases.size === 0 &&
-      this.module === "default"
+      this.module === "default" &&
+      this.transactionOptions === TransactionOptions.defaults() &&
+      this.retryOptions === RetryOptions.defaults()
```

Comment:
@1st1 I had to add the `transactionOptions` check here to get it to hit the state serialization path since you changed all of this code in your codec work. Do you mind just giving this method another close look to make sure we're not missing other places where state isn't being updated?

Reply:
Hm... I can try.

Separately speaking, why are we checking `retryOptions`? I don't think it affects server state in any way, it should be a pure client-side thing.

Reply:
I mostly just wanted to make sure we were checking _all_ options to know if we should go into this faster path. I noticed that we don't have this fast path in Python, so I thought it would be safer to stick to a stricter sense of "default" here, but it's definitely true that this doesn't affect the state, so I'm happy to drop it.